apps/traincascade/old_ml_tree.cpp

```diff
@@ -3798,59 +3807,59 @@ void CvDTree::write_split( CvFileStorage* fs, CvDTreeSplit* split ) const
         // to achieve more compact and clear representation
         default_dir = to_right <= 1 || to_right <= MIN(3, n/2) || to_right <= n/3 ? -1 : 1;
 
-        cvStartWriteStruct( fs, default_dir*(split->inversed ? -1 : 1) > 0 ?
-                            "in" : "not_in", CV_NODE_SEQ+CV_NODE_FLOW );
+        fs.startWriteStruct( default_dir*(split->inversed ? -1 : 1) > 0 ?
+                             "in" : "not_in", FileNode::SEQ+FileNode::FLOW );
 
         for( i = 0; i < n; i++ )
         {
             int dir = CV_DTREE_CAT_DIR(i,split->subset);
             if( dir*default_dir < 0 )
-                cvWriteInt( fs, 0, i );
+                fs.write( 0, i );
         }
-        cvEndWriteStruct( fs );
+        fs.endWriteStruct();
     }
     else
-        cvWriteReal( fs, !split->inversed ? "le" : "gt", split->ord.c );
+        fs.write( !split->inversed ? "le" : "gt", split->ord.c );
 
-    cvEndWriteStruct( fs );
+    fs.endWriteStruct();
 }
 
 
-void CvDTree::write_node( CvFileStorage* fs, CvDTreeNode* node ) const
+void CvDTree::write_node( cv::FileStorage& fs, CvDTreeNode* node ) const
 {
     CvDTreeSplit* split;
 
-    cvStartWriteStruct( fs, 0, CV_NODE_MAP );
+    fs.startWriteStruct( 0, FileNode::MAP );
 
-    cvWriteInt( fs, "depth", node->depth );
-    cvWriteInt( fs, "sample_count", node->sample_count );
-    cvWriteReal( fs, "value", node->value );
+    fs.write( "depth", node->depth );
+    fs.write( "sample_count", node->sample_count );
+    fs.write( "value", node->value );
 
     if( data->is_classifier )
-        cvWriteInt( fs, "norm_class_idx", node->class_idx );
+        fs.write( "norm_class_idx", node->class_idx );
 
-    cvWriteInt( fs, "Tn", node->Tn );
-    cvWriteInt( fs, "complexity", node->complexity );
-    cvWriteReal( fs, "alpha", node->alpha );
-    cvWriteReal( fs, "node_risk", node->node_risk );
-    cvWriteReal( fs, "tree_risk", node->tree_risk );
-    cvWriteReal( fs, "tree_error", node->tree_error );
+    fs.write( "Tn", node->Tn );
+    fs.write( "complexity", node->complexity );
+    fs.write( "alpha", node->alpha );
+    fs.write( "node_risk", node->node_risk );
+    fs.write( "tree_risk", node->tree_risk );
+    fs.write( "tree_error", node->tree_error );
 
     if( node->left )
     {
-        cvStartWriteStruct( fs, "splits", CV_NODE_SEQ );
+        fs.startWriteStruct( "splits", FileNode::SEQ );
 
         for( split = node->split; split != 0; split = split->next )
```

Comment:
`for( CvDTreeSplit* split = node->split; split != 0; split = split->next )` and remove `CvDTreeSplit* split;` definition.

Reply:
ok